replica/backup.go

```diff
@@ -270,8 +275,12 @@ func preload(d *diffDisk) error {
 					length++
 				}
 			}
-			d.location[offset] = byte(i)
-			d.UsedBlocks++
+			d.rmLock.Lock()
+			if d.location[offset] < byte(len(d.files)) {
```

Comment:
is this supposed to be d.location[offset] < (byte(len(d.files)) -1 )

Reply:
d.location[offset] != (byte(len(d.files))) is correct validation, explanation is as follows:
- d.files[0] = <nil>
- d.files[1]  = file1 
and so on
on each restart a new file get appended to d.files so the latest file will be at the last location. For example if there was 5 files initially and restart happens so total file is 5+1(new file) = 6 and size of d.files would be 6 + 1(<nil> oth location) = 7. So we will verify that if offset is in latest file or not, if it's already  there then we will ignore and if it's not there will update it.